src/lwc/utilCustomErrors/exceptionDataError.js

```diff
@@ -9,22 +10,16 @@ class ExceptionDataError extends Error {
     */
     constructor(apexException) {
         super();
-        if (apexException.body && apexException.body.exceptionType) {
```

Comment:
@dfuller114 I'm worried about removing this block - before your updates here, this method looks like it was built to handle exceptions that were provisioned with serialized ERR_ExceptionData (which you've preserved), and also handle exceptions in their raw format, which was handled in this block that's being removed. How sure are we that we don't have to handle basic, "untreated" exceptions?

Reply:
@ptewson-sfdo the idea was to simplify the logic here and remove duplicate logic since it doesn't seem like these two blocks are necessary. We are populating the majority of all the properties of the `Error` parent in both blocks, which will then be handled in whatever LWC component consumes it. In the second block, the `message` property is parsed which is what the `validateJSONString` function does if the string is valid JSON. So it seemed cleaner to have the first block populate all the properties if we have a `body` and `exceptionType` (whether there are values there or not), then have the second block directly set the `apexException`  to the `message` property if that scenario applies which is a different scenario. The `if, if/else` blocks don't seem different enough to warrant an entirely separate block. @psadie-sfdo you might can confirm. think we discussed this some when I made the change.

Reply:
I think that as this is written, it will choke on exceptions that haven't been provisioned with the serialized content in the message (validateJSONString() will return `false` and then the attempts to reference properties of apexExceptionWrapper will fail. I think we an protect against that, however, by re-writing this block (below the call to super()) with something like: 
```

       const apexExceptionWrapper = validateJSONString(apexException.body.message);
        if (apexExceptionWrapper === false) {
            this.exceptionType = apexException.body.exceptionType;
            this.errorMessage = apexException.body.message;
            this.stackTrace = apexException.body.stackTrace;
            this.isUserDefinedException = apexException.body.isUserDefinedException;
            this.DMLErrorMessageMapping = {};
            this.DMLErrorFieldNameMapping = {};

        } else if (apexException.body && apexException.body.message) {
            // This looks like an instance of the apex wrapper class ERR_ExceptionData
            this.exceptionType = apexExceptionWrapper.exceptionType;
            this.errorMessage = apexExceptionWrapper.errorMessage;
            this.DMLErrorMessageMapping = apexExceptionWrapper.DMLErrorMessageMapping;
            this.DMLErrorFieldNameMapping = apexExceptionWrapper.DMLErrorFieldNameMapping;

        } else {
            this.message = apexException;
        }
```

Unfortunately, I think based on the conversation with the team we should roll back the rest of the changes here for the time being. I think we've collectively identified a need for refactor with error handling (and de-prioritized the field-level messaging, which I think introduced a lot of the complexity in the first place), but we need to approach that with caution and enough time to deliver that change thoughtfully. For this Work Item we probably need to focus on the updates that are constrained to closing the bug that originated this work - which I believe the suggested change here does.